examples/nary.mcl

```diff
@@ -0,0 +1,49 @@
+(* main body *)
+(
+  fun (Nary : forall (n : Nat) -> Type@0)
+      (appNary : forall (n : Nat) (f : Nary (succ n)) (arg : Nat) -> Nary n)
```

Comment:
I would swap the order of this input and the next, just to line up the order below in the recursion.